### PR TITLE
Various enhancements to nut package

### PR DESCRIPTION
**Pull request by dfc-d**
@cyanidium 

This is in middle of my testing/debugging, but I wanted to run this set of changes past you (for your opinion) before declaring it finished (once tested) and taking over maintership as per our emails.

The pull request has a number of different changes, since I was interested in quite a lot of stuff for this.  Do you think I should break it into separate PR's when I submit, or is a single PR fine?

### CODE REVIEW

**Comment by cyanidium** [comment]
Wow, you weren't joking about about all the features you wanted to bring in. Nice work! My $0.02:

- I would have 1 PR with all of the changes. The only issue with that is the change to collectd, but it fixes a problem your changes introduce by renaming packages, so getting it all at once makes sense to me.

- I would leave serial in there (both the drivers and building with serial support). There isn't really any overhead to having it as an option and while serial ports are rare, they are still on new devices. Unlike USB and SNMP there isn't a dependency required, and the code overhead is negligable (last I checked it wasn't actually anything). This way if someone has a serial device, they just need to install the driver and it will work. People without the driver don't suffer for it either because the package size barely changes.

- The indentation in the Makefile is inconsistent. For example, many $(call ...)s are indented differently than the rest of the section, Default and nut-web-cgi descriptions are not indented but others are, and Package/nut-upsmon-sendmail-notify is less indented than other Package sections. Functionally it works the same, but with large changes to the file it does make it hard to follow and will make future PRs harder to read because of whitespace changes.

- Group nut and nut-common sections separately. At the moment they are mixed together, mostly because the nut-common Package is all the way at the start.

Otherwise I like the work you have done, ucifying will be a big improvement and the package name changes will simplify things.

**Comment by dfc-d** [comment]
Hi,

I've reworked as you suggested.

* whitespace consistent (and consistent with previous usage)
* nut-common moved - it was a misguided attempt to avoid extra lines appearing in the diff
* I change the remove serial to fix missing pieces for serial to work (I guess you had accidentally dropped some pieces needed for serial).

Now I just need to retest.

**Comment by dfc-d** [comment]
@cyanidium 

Ok, I've finally got this PR tested with the proposed changes, and it's working like a charm (at least for me).  I'd love your feedback, and hopefully your ACK on the changes including maintership!

**Comment by cyanidium** [comment]
It tests fine for me, and I've got no additional feedback other than to say: awesome work!

Recommend it gets merged.

File: utils/collectd/Makefile [hunk 1]
```diff
@@ -335,7 +335,7 @@ $(eval $(call BuildPlugin,netlink,netlink input,netlink,+PACKAGE_collectd-mod-ne
 $(eval $(call BuildPlugin,network,network input/output,network,+PACKAGE_COLLECTD_ENCRYPTED_NETWORK:libgcrypt))
 $(eval $(call BuildPlugin,nginx,nginx status input,nginx,+PACKAGE_collectd-mod-nginx:libcurl))
 $(eval $(call BuildPlugin,ntpd,NTP daemon status input,ntpd,))
-$(eval $(call BuildPlugin,nut,UPS monitoring input,nut,+PACKAGE_collectd-mod-nut:nut))
+$(eval $(call BuildPlugin,nut,UPS monitoring input,nut,+PACKAGE_collectd-mod-nut:nut-common))
```
Comment (champtar) [inline utils/collectd/Makefile]: you may need to bump collectd PKG_RELEASE

File: net/nut/files/nut-monitor.init [hunk 2]
```diff
@@ -159,6 +159,11 @@ start_service() {
 	chmod 640 $UPSMON_C
 	chmod 640 /var/etc/nut/nut.conf
 
+	[ -d /var/run/nut ] || {
+		mkdir -m 0750 -p /var/run/net
```
Comment (yousong) [inline net/nut/files/nut-monitor.init]: Typo in `/var/run/net`?

File: net/nut/files/nut-monitor.init [hunk 3]
```diff
@@ -2,18 +2,175 @@
...
+	    system="$system:$port";
+	fi
+	echo "MONITOR $system $powervalue $username $password $type" >> $UPSMON_C
 }
 
 start_service() {
-	upsmon -p
+	[ -d $RUN_D ] || mkdir -p $RUN_D
+        mkdir -p /var/etc/nut
+        chmod 640 /var/etc/nut
```
Comment (yousong) [inline net/nut/files/nut-monitor.init]: The above 2 lines serve the same purposes as the `[ -d /var/run/nut ]` check below?  and they have blankspaces instead of tabs for indentation.

File: net/nut/files/nut-sendmail-notify [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+(cat - | /usr/sbin/sendmail root ) <<EOF
+From: root
+To: root
+Subject: UPS $NOTIFYTYPE Notification
+
+$1
+.
+EOF
```
Comment (yousong) [inline net/nut/files/nut-sendmail-notify]: Guess there is no need to use subshell.  I see no reason why sendmail cannot work directly with here document ;)

**Comment by dfc-d** [comment]
@cyanidium 
Suggested fixes pushed will let you know when I've tested (Yousong makes good suggestions).

File: net/nut/files/nut-sendmail-notify (hunk 4 above)
Comment (dfc-d) [inline net/nut/files/nut-sendmail-notify]: On Mon, 2016-07-04 at 06:14 -0700, Yousong Zhou wrote:
[various fixes]

Thank you for your input; that's the kind of thing testing doesn't
really catch but is good to do right.

I'll update the things you pointed out.

Regards,

Daniel

File: utils/collectd/Makefile (hunk 1 above)
Comment (dfc-d) [inline utils/collectd/Makefile]: On Mon, 2016-07-04 at 05:40 -0700, champtar wrote:
> input,nut,+PACKAGE_collectd-mod-nut:nut-common))
> you may need to bump collectd PKG_RELEASE

In fact, yes, I missed that.  Thanks.

Regards,

Daniel

File: net/nut/files/nut-monitor.init (hunk 2 above)
Comment (dfc-d) [inline net/nut/files/nut-monitor.init]: On Mon, 2016-07-04 at 06:03 -0700, Yousong Zhou wrote:
> In net/nut/files/nut-monitor.init:
> > @@ -159,6 +159,11 @@ start_service() {
> >  	chmod 640 $UPSMON_C
> >  	chmod 640 /var/etc/nut/nut.conf
> >  
> > +	[ -d /var/run/nut ] || {
> > +		mkdir -m 0750 -p /var/run/net
> Typo in /var/run/net?

I think this one didn't trigger because I was also running nut-server
which already created it.

Thanks,

Daniel

File: net/nut/files/nut-sendmail-notify (hunk 4 above)
Comment (dfc-d) [inline net/nut/files/nut-sendmail-notify]: On Mon, 2016-07-04 at 06:14 -0700, Yousong Zhou wrote:
> In net/nut/files/nut-sendmail-notify:
> > @@ -0,0 +1,10 @@
> > +#!/bin/sh
> > +
> > +(cat - | /usr/sbin/sendmail root ) <<EOF
> > +From: root
> > +To: root
> > +Subject: UPS $NOTIFYTYPE Notification
> > +
> > +$1
> > +.
> > +EOF
> Guess there is no need to use subshell. I see no reason why sendmail
> cannot work directly with here document ;)

Erm, yeah.  Do I get a useless use of cat award ;-) ?

Regards,

Daniel

File: net/nut/Makefile [hunk 5]
```diff
@@ -9,28 +9,22 @@ include $(TOPDIR)/rules.mk
...
-	CONFIG_NUT_CLIENTS_UPSC \
-	CONFIG_NUT_CLIENTS_UPSCMD \
-	CONFIG_NUT_CLIENTS_UPSLOG \
-	CONFIG_NUT_CLIENTS_UPSMON \
-	CONFIG_NUT_CLIENTS_UPSRW \
-	CONFIG_NUT_CLIENTS_UPSSCHED \
 	CONFIG_NUT_DRIVER_SNMP \
 	CONFIG_NUT_DRIVER_USB \
-	CONFIG_NUT_SERVER \
+	CONFIG_NUT_DRIVER_SERAL \
```
Comment (phillxnet) [inline net/nut/Makefile]: @dfc-d I suspect CONFIG_NUT_DRIVER_SERAL should be CONFIG_NUT_DRIVER_SERIAL.
typo?
Thanks for all your work on this by the way.

File: net/nut/Makefile [hunk 6]
```diff
@@ -52,68 +46,240 @@ endef
...
+
+define Package/nut-upsmon-sendmail-notify/description
+$(call Package/nut/description/Default)
+upsmon with default notification via sendmail
+endef
+
+define Package/nut-upsmon-sendmail-notify/install
+	$(INSTALL_DIR) $(1)/usr/bin $(1)/etc/uci-defaults
+	$(INSTALL_BIN) ./files/nut-sendmail-notify $(1)/usr/bin/
+	$(INSTALL_DATA) ./files/nut-sendmail-notify.default $(1)/etc/uci-defaults/nut-sendmail-notify
```
Comment (phillxnet) [inline net/nut/Makefile]: @dfc-d I believe the 2 instances of INSTALL_DATA in the Makefile should be INSTALL_BIN as the resulting /etc/uci-defaults files (nut-sendmail-notify and nut-upssched) are required to have execution permission, otherwise they fail to run but are deleted anyway. I have only tested this for the /etc/uci-defaults/nut-sendmail-notify file though).
Apologies if I have this wrong, my tests are of this pr build within the 15.05 branch.

**Comment by phillxnet** [comment]
@dfc-d Are we not missing the libpthread dependency from the nut-common section?

File: net/nut/Config.in [hunk 7]
```diff
@@ -97,4 +23,10 @@
 		help
 			If you have a UPS you can connect to via SNMP, select this.
 		default n
-		
+
+	config NUT_DRIVER_SERIAL
+		depends on PACKAGE_nut
+		bool "Build with suport for serial drivers"
```
Comment (phillxnet) [inline net/nut/Config.in]: Typo in menuconfig facing text "suport" ie support

**Comment by phillxnet** [comment]
@dfc-d Looks like we are missing the following line in the nut-web-cgi Package section to install the upsset.conf from it's sample:
```
$(INSTALL_CONF) $(PKG_INSTALL_DIR)/etc/nut/upsset.conf.sample $(1)/etc/nut/upsset.conf
```

File: net/nut/Makefile (hunk 6 above)
Comment (dfc-d) [inline net/nut/Makefile]: @philxnet uci-defaults is sourced and therefore doesn't need to be executable (/lib/preinit is the same way).  If nut-sendmail-notify default script is not working is almost certainly for some other reason (and it works for me).

File: net/nut/Config.in (hunk 7 above)
Comment (dfc-d) [inline net/nut/Config.in]: Does build fail for you without libpthread dependency?  I'm not sure it is actually required (I've had no issues but I primarily use MUSL based builds so maybe GLIBC requires it?

And typo noted.

**Comment by dfc-d** [comment]
I didn't include upsset because I hadn't ucified upsset config and nut-web-cgi is intended to be an uci based (for the LuCI app I've written) configuration.  I suppose I could include that file for those that want upsset and don't want for however long it will take to get to ucification.

**Comment by dfc-d** [comment]
@cyanidium 
Hi finally have this tested again, and it seem to be working well.  ACK to transfer maintainership and go ahead with this patch?  I haven't included upsset yet.  @phillxnet If you want to submit a patch and make sure it works, then that would speed things up.

**Comment by cyanidium** [comment]
Looks good to me, it's all yours.

**Comment by dfc-d** [comment]
need to resolve conflict due to collectd update.
